gplearn/genetic.py

```diff
@@ -441,20 +462,24 @@ def fit(self, X, y, sample_weight=None):
             for program in population:
                 program.fitness_ = program.fitness(parsimony_coefficient)
 
-            self._programs.append(population)
+            if self.slim:
+                self._parent_generation = population
+            else:
```

Comment:
While I get what you are looking to achieve, this will break warm_start. I think the best solution is to maintain all the generational stuff in the self._program list and instead null out the earlier generations based on the option chosen. That way the user's interaction with the final generation is unchanged, the memory consumption should be close to what you see in this PR's version, and documentation will be similar no matter what option you choose. Great work so far. I like it. And thank you